hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -682,26 +694,27 @@ update ::
   IsChainState tx =>
   Environment ->
   Ledger tx ->
+  Word64 ->
   -- | Current HeadState to validate the command against.
   HeadState tx ->
   -- | Command sent to the HeadLogic to be processed.
   Event tx ->
   Outcome tx
-update env ledger st ev = case (st, ev) of
+update env ledger nextStateChangeID st ev = case (st, ev) of
```

Comment:
Must: not change the `HeadLogic.update` function.

This represents the heart of the protocol logic, and nothing here should require knowledge of what the next state changed event identifier could be.

I'll comment in `Hydra.Node` what our options are.